app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt

```diff
@@ -363,14 +363,21 @@ class AccountsFragment : BaseFragment(), OnRefreshListener, AccountsView {
      */
     fun filterSavingsAccount(statusModelList: List<CheckboxStatus?>?) {
         val filteredSavings: MutableList<SavingAccount?> = ArrayList()
-        if (accountsPresenter?.getCheckedStatus(statusModelList) != null && accountsPresenter != null) {
+        if (accountsPresenter?.getCheckedStatus(statusModelList) != null && accountsPresenter != null&&accountsPresenter?.getCheckedStatus(statusModelList)!!
+                .isNotEmpty()) {
```

Comment:
use safe call operator instead of not-null assertion operator

Reply:
Already we checked for null at the starting of the if statement so we can use a not null assertion.

Reply:
you are right but using the safe call operator again provides an additional layer of null safety, ensuring that the subsequent method call is only executed if accountsPresenter is not null.